### ISSUE TITLE
`_TensorDict.permute(...)` operation (+ transpose)

### ISSUE THREAD

**Issue by vmoens** [facebookresearch/rl#59]
The `_TensorDict` parent class supports a bunch of tensor operations such as `view` and `unsqueeze`.
Modifications of those tensordicts will result in changes in the original tensordict (provided they don't violate the tensordict specs). 

`unsqueeze`:
```python
>>> t = TensorDict({'a': torch.randn(3, 4)}, [3])
>>> t.unsqueeze(0).set('b', torch.randn(1, 3, 4));
>>> print(t)
TensorDict(
    fields={
        a: Tensor(torch.Size([3, 4]), dtype=torch.float32),
        b: Tensor(torch.Size([3, 4]), dtype=torch.float32)},
    batch_size=torch.Size([3]),
    device=cpu,
    is_shared=False)
>>> t.unsqueeze(0).fill_('a', 0.0);
>>> print(t.get('a'))
tensor([[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]])
```

`view`:
```python
>>> t = TensorDict({'a': torch.randn(3, 4)}, [3])
>>> t.view(-1).set('b', torch.randn(12));
>>> print(t)
TensorDict(
    fields={
        a: Tensor(torch.Size([3, 4]), dtype=torch.float32),
        b: Tensor(torch.Size([3, 4]), dtype=torch.float32)},
    batch_size=torch.Size([3]),
    device=cpu,
    is_shared=False)
>>> t.view(-1).fill_('a', 0.0);
>>> print(t.get('a'))
tensor([[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]])
```

A similar behaviour should be implemented for `_TensorDict.permute(...)`. The expected behaviour would be:
`view`:
```python
>>> t = TensorDict({'a': torch.randn(3, 4, 1)}, [3, 4])
>>> t.permute(1, 0).set('b', torch.randn(4, 3));
>>> print(t)
TensorDict(
    fields={
        a: Tensor(torch.Size([3, 4]), dtype=torch.float32),
        b: Tensor(torch.Size([3, 4]), dtype=torch.float32)},
    batch_size=torch.Size([3]),
    device=cpu,
    is_shared=False)
>>> t.permute(1, 0).fill_('a', 0.0);
>>> print(t.get('a'))
tensor([[[0.],
         [0.],
         [0.],
         [0.]],
        [[0.],
         [0.],
         [0.],
         [0.]],
        [[0.],
         [0.],
         [0.],
         [0.]]])
```

The same should work with transpose (which would permute two dimensions only).

This would require a new `_CustomOpTensorDict` subclass, similar to `ViewedTensorDict` and `UnsqueezedTensorDict`.

**Comment by vmoens** [comment]
Closed by #62

**Comment by vmoens** [comment]
closed as per #99 
thanks @TimoStoff
